server/src/main/java/org/elasticsearch/rest/CompatibleHandlers.java

```diff
@@ -0,0 +1,69 @@
+package org.elasticsearch.rest;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.Version;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.index.mapper.MapperService;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.UnaryOperator;
+
+public class CompatibleHandlers {
```

Comment:
Can you add a TODO in here to ensure we get some buy in on https://github.com/elastic/elasticsearch/issues/52370 before we merge the header format.